test/alphabet/alphabet_test.cpp

```diff
@@ -71,7 +73,7 @@ TYPED_TEST(alphabet, assign_rank)
     EXPECT_EQ((assign_rank(TypeParam{}, 0)), TypeParam{});
 
     TypeParam t0;
-    for (typename TypeParam::rank_type i = 0; i < alphabet_size_v<TypeParam>; ++i)
+    for (uint64_t i = 0; i < alphabet_size_v<TypeParam>; ++i)
```

Comment:
cool, now we can add `gap` as typed test. Because `false++` is forbidden since `c++17` and I couldn't add it in the list above.